ledger/blockqueue_test.go

```diff
@@ -0,0 +1,81 @@
+// Copyright (C) 2019-2021 Algorand, Inc.
+// This file is part of go-algorand
+//
+// go-algorand is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as
+// published by the Free Software Foundation, either version 3 of the
+// License, or (at your option) any later version.
+//
+// go-algorand is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with go-algorand.  If not, see <https://www.gnu.org/licenses/>.
+
+package ledger
+
+import (
+	"fmt"
+	"github.com/algorand/go-algorand/protocol"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+
+	"github.com/algorand/go-algorand/agreement"
+	"github.com/algorand/go-algorand/config"
+	"github.com/algorand/go-algorand/crypto"
+	"github.com/algorand/go-algorand/data/bookkeeping"
+	"github.com/algorand/go-algorand/logging"
+)
+
+func TestPutOldBlock(t *testing.T) {
+	genesisInitState, _, _ := genesis(10)
+
+	dbName := fmt.Sprintf("%s.%d", t.Name(), crypto.RandUint64())
+	const inMem = true
+	cfg := config.GetDefaultLocal()
+	cfg.Archival = true
+	l, err := OpenLedger(logging.Base(), dbName, inMem, genesisInitState, cfg)
+	require.NoError(t, err)
+	defer l.Close()
+
+	blk := bookkeeping.Block{}
+	var cert agreement.Certificate
+	err = l.blockQ.putBlock(blk, cert)
```

Comment:
Please do something more meaningful, like checking that the opearion truely did what it was supposed to do..

Reply:
Sorry, the purpose of this test was only to make sure that `putBlock` would error when the block put was too old. Perhaps the name of the test should have been made more clear to convey that.

Reply:
ah - so, in that case,  please make sure you're testing a specific error rather than checking the existence of an error.